### PR TITLE
Generate encrypted image

### PR DESCRIPTION
**Pull request by BShrivastav**
Now user of the IBM Cloud Plugin can generate encrypted images using a base image.
Use `encryption_key_crn` for same.

### CODE REVIEW

File: examples/build.vpc.zprofile.pkr.hcl [hunk 1]
```diff
@@ -7,6 +7,10 @@ packer {
   }
 }
 
+variable "ENCRYPTION_KEY_CRN" {
```
Comment (astha-jain) [inline examples/build.vpc.zprofile.pkr.hcl:10]: Following the other examples, version change is reqd line 4 in this file.

File: examples/build.vpc.centos-parallel.pkr.hcl [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+      version = ">=v3.0.4"
       source = "github.com/IBM/ibmcloud"
     }
   }
 }
 
+variable "ENCRYPTION_KEY_CRN" {
+  type = string
+}
+
```
Comment (astha-jain) [inline examples/build.vpc.centos-parallel.pkr.hcl:13]: ENCRYPTION_KEY_CRN is not being used anywhere. Do we need this here. Same for all files except encryption example

**Review by astha-jain — COMMENTED** [review]
Tested. works well. Please address the comments and this should be good to go.

File: README.md [hunk 3]
```diff
@@ -199,6 +199,7 @@ vsi_user_data_file | string | User data to be made available when setting up the
 vpc_endpoint_url | string | Configure URL for VPC test environments. Optional.
 iam_url | string | Configure URL for IAM test environments. Optional.
 image_name | string | The name of the resulting custom Image that will appear in your account. Required.
+encryption_key_crn | string | The crn of the key used for encryption, from key mangement service of IBM Cloud like Hyper Protect or [Key Protect](https://cloud.ibm.com/docs/key-protect?topic=key-protect-about)
```
[the remark this answers is not in the document]
Comment (BShrivastav) [reply README.md]: This is done.

File: examples/build.vpc.centos-parallel.pkr.hcl (hunk 2 above)
Comment (BShrivastav) [reply examples/build.vpc.centos-parallel.pkr.hcl:13]: It was kept as a placeholder, will remove it. Make sense to just keep it at one example.

File: developer/variables.pkrvars.hcl [hunk 4]
```diff
@@ -2,4 +2,5 @@ SUBNET_ID = " "
 REGION = " "
 SECURITY_GROUP_ID = " "
 RESOURCE_GROUP_ID = " "
-IBM_API_KEY = " "
\ No newline at end of file
+IBM_API_KEY = " "
+ENCRYPTION_KEY_CRN = ""
```
Comment (astha-jain) [inline developer/variables.pkrvars.hcl:6]: ` ENCRYPTION_KEY_CRN = ""` has to be `ENCRYPTION_KEY_CRN = " "`
`""` messes up with some build I am hard time trying to recall

Comment (BShrivastav) [reply developer/variables.pkrvars.hcl]: addressed.

**Review by astha-jain — APPROVED** [review]
Looks good.

**Review by sumitkumartiwari — APPROVED** [review]
Looks good..

**Pull request merged**